### PR TITLE
reduce a lot of final inference GPU memory

### PR DESCRIPTION
**Pull request by Johnson-yue**
# What does this PR do?

I remove parameters and models in training, and those not use when final inference. It can be saved a lot of GPU memory for final inference

### CODE REVIEW

File: examples/dreambooth/train_dreambooth_lora_sdxl.py [hunk 1]
```diff
@@ -1677,6 +1677,13 @@ def compute_text_embeddings(prompt, text_encoders, tokenizers):
             text_encoder_2_lora_layers=text_encoder_2_lora_layers,
         )
 
+        # Remove unuse parameters and models to save a lot of GPU memory for Final inference
+        unet = unet.cpu()
```
Comment (sayakpaul) [inline examples/dreambooth/train_dreambooth_lora_sdxl.py:1681]: Is this placement necessary?

File: examples/dreambooth/train_dreambooth_lora_sdxl.py [hunk 2]
```diff
@@ -1677,6 +1677,13 @@ def compute_text_embeddings(prompt, text_encoders, tokenizers):
...
+        text_encoder_one = text_encoder_one.cpu()
+        text_encoder_two = text_encoder_two.cpu()
+        del unet,text_encoder_one, text_encoder_two
+        del optimizer
```
Comment (sayakpaul) [inline examples/dreambooth/train_dreambooth_lora_sdxl.py:1685]: Can be ammend in the LoC before.

Comment (Johnson-yue) [reply examples/dreambooth/train_dreambooth_lora_sdxl.py:1685]: Maybe is ok。 I write this for remove unuseful GPU tensors and models 。 So， if any code will implement this function，will can be

Comment (sayakpaul) [reply examples/dreambooth/train_dreambooth_lora_sdxl.py:1685]: I meant doing something like: `del unet, text_encoder_one, text_encoder_two, optimizer`.

And if we do that I don't think doing `.cpu()` calls to the text encoders will be necessary.

Comment (Johnson-yue) [reply examples/dreambooth/train_dreambooth_lora_sdxl.py:1685]: I test directly  `del net, text_encoder_one, text_encoder_two, optimizer` but GPU memory not release, I don't know why. So I convert those model to cpu device and then delete them, and it work for me

**Pull request closed without merging**